### DISCUSSION TITLE
Feature request: throttling mechanism?

### DISCUSSION THREAD

**Discussion by wimpie3** [jchristn/SuperSimpleTcp#55] [Ideas]
Just thinking out loud here... I'm currently using your library to simulate a slow modem connection. I'm doing this by using a timer and sending out one character at a time, which makes the characters appear slowly on the other side. This, however, requires me to create my own in/out buffers inside my program, on top of those that exist in simpletcp.

I'm not sure if I'm the only one with this need but some kind of throttling mechanism that limits the speed of the data that is being transferred would be a cool addition to simpletcp (something like: "limit output to 100 bytes per second" or so...)

**Comment by jchristn** [comment]
Hi @wimpie3 definitely out of scope for me (have had this request for another TCP library I manage, too).  However you if put together a PR on it that cleanly integrates I'd love to take a look.
